corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -37,35 +37,6 @@ def validate_phone_datetime(datetime_string, none_ok=False, form_id=None):
         raise PhoneDateValueError('{!r}'.format(datetime_string))
 
 
-def post_case_blocks(case_blocks, form_extras=None, domain=None, user_id=None, device_id=None):
-    """
-    Post case blocks.
-
-    Extras is used to add runtime attributes to the form before
-    sending it off to the case (current use case is sync-token pairing)
-
-    See `device_id` parameter documentation at
-    `corehq.apps.hqcase.utils.submit_case_blocks`.
-    """
-    from corehq.apps.hqcase.utils import submit_case_blocks
-
-    if form_extras is None:
-        form_extras = {}
-
-    domain = domain or form_extras.pop('domain', None)
-    if getattr(settings, 'UNIT_TESTING', False):
-        from casexml.apps.case.tests.util import TEST_DOMAIN_NAME
-        domain = domain or TEST_DOMAIN_NAME
-
-    return submit_case_blocks(
-        [ElementTree.tostring(case_block, encoding='utf-8').decode('utf-8') for case_block in case_blocks],
```

Comment:
It seems like `post_case_blocks` does some defensive parameter stuff, I guess mainly looking for domain in either the `domain` param or `form_extras`. Were you able to confirm it's impossible for `domain` to be `None` in any call to this? I couldn't find where that logic was replicated elsewhere, but I might have missed it.

Reply:
(My review here isn't blocking by the way.)

Reply:
The domain default is purely for testing convenience. It is a requirement param for SubmissionPost: https://github.com/dimagi/commcare-hq/blob/ce871c5ccd0525becb5bdc145fd9cfd89547b359/corehq/form_processor/submission_post.py#L73

I would actually prefer to have `domain` always be an explicitly required arg rather than have a default. I may PR a change for that later.